docs/tutorials/shallow-join.rst

```diff
@@ -0,0 +1,359 @@
+an Array before broadcasting. Spark does not allow broadcasting of RDDs, and due to the small size of the data set, we
+can safely collect data onto the driver node without a risk of running out of memory.
+
+.. code-block:: scala
+
+    val broadcastedRegions = sc.broadcast(countriesRdd.collect)
+
+With the covering set available on each partition, we can iterate over the GDELT events and key them by the region they
+were contained in. In ``mapPartitions``, ``iter`` is an iterator to all the elements (in this case Simple Features) on
+the partition. Here we transform each iterator and store the result into a new RDD.
+
+.. code-block:: scala
+
+    val keyedData = gdeltRdd.mapPartitions { iter =>
+        import org.locationtech.geomesa.utils.geotools.Conversions._
+
+        iter.flatMap { sf =>
+            // Iterate over regions until a match is found
+            val it = broadcastedRegions.value.iterator
+            var container = ""
+            while (it.hasNext() && container == "") {
+                val country = it.next()
+                // If the polygon contains the event, set the container
+                if (country.geometry.contains(sf.geometry)) {
+                    container = country.getAttribute("NAME").asInstanceOf[String]
+                }
+            }
+            // return the found country as the key
+            if (container != "") {
+                Some(container, sf)
+            } else {
+                None
+            }
+        }
+    }
+
+Our new RDD is now of type ``RDD[(String, SimpleFeature)]`` and can be used for a Spark ``reduceByKey`` operation, but
+first, we need to create a simple feature type to represent the aggregated data.
+
+Creating a New Simple Feature Type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We first loop through the attributes of a sample feature from the GDELT RDD to decide what fields can be aggregated.
+
+.. code-block: scala
+
+    val featureAttributes = gdeltRdd.first.getAttributes.toSeq
+    val countableIndices = featureAttributes.toIndexedSeq.indices.flatMap( { index =>
+        // Skip the id
+        if (index != 0) {
+            featureAttributes(index) match {
+                case attr if attr != null && attr.getClass == classOf[Integer] =>
+                    Some(index, "Integer")
+                case attr if attr != null && attr.getClass == classOf[java.lang.Long] =>
+                    Some(index, "Long")
+                case attr if attr != null && attr.getClass == classOf[java.lang.Double] =>
+                    Some(index, "Double")
+                case _ => None
+            }
+        } else {
+            None
+        }
+    }).toArray
+    val countable = sc.broadcast(countableIndices)
+
+With these fields, we can create a Simple Feature Type to store their averages and totals, prefixing each one with
+"total_" and "avg_". Of course, it may not make sense to aggregate features like "ID" should it appear, but this
```

Comment:
We have warnings for that; check out https://github.com/locationtech/geomesa/blob/master/geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/GeoMesaSchemaValidator.scala.;)